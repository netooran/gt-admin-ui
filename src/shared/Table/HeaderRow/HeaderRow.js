import PropTypes from 'prop-types';
import { useEffect, useState } from 'react';
import { FunctionUtils } from 'utils/utils/utils';

const HeaderRow = ({ options, columns, isSelected, selectAll }) => {
  const [isChecked, setIsChecked] = useState(isSelected);
  const { selectable, ...restOfOptions } = options;

  const hasActions = !!Object.values(restOfOptions).some(
    FunctionUtils.identity
  );

  useEffect(() => {
    setIsChecked(isSelected);
  }, [isSelected]);

  return (
    <tr>
      {selectable && (
        <th>
          <input
            type="checkbox"
            aria-label="Select all"
            checked={isChecked}
            onChange={(event) => {
              setIsChecked(event.target.checked);
              selectAll(event.target.checked);
            }}
          />
        </th>
      )}
      {columns.map((column, i) => (
        <th key={i}>{column}</th>
      ))}
      {hasActions && <th>actions</th>}
    </tr>
  );
};

HeaderRow.propTypes = {
  columns: PropTypes.array.isRequired,
  isSelected: PropTypes.bool,
  selectAll: PropTypes.func,
  options: PropTypes.shape({
    selectable: PropTypes.bool,
  }),
};

export default HeaderRow;
