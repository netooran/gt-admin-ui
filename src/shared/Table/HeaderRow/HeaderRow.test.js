import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import HeaderRow from './HeaderRow';

test('renders header row with select all', () => {
  const props = {
    options: { selectable: true },
    columns: ['name', 'email'],
    isSelected: false,
    selectAll: jest.fn(),
  };

  const { container, getByRole } = render(<HeaderRow {...props} />);

  expect(container).toMatchSnapshot();

  userEvent.click(getByRole('checkbox'));
  expect(getByRole('checkbox')).toBeChecked();
  expect(props.selectAll).toHaveBeenLastCalledWith(true);
});

test('renders header row without select all', () => {
  const props = { options: { selectable: false }, columns: ['name', 'email'] };
  const { container } = render(<HeaderRow {...props} />);
  expect(container).toMatchSnapshot();
});

test('renders header row with actions', () => {
  const props = {
    options: { selectable: false, editable: true, deletable: false },
    columns: ['name', 'email'],
  };
  const { container } = render(<HeaderRow {...props} />);
  expect(container).toMatchSnapshot();
});

test('renders header row without actions', () => {
  const props = {
    options: { selectable: false, editable: false },
    columns: ['name', 'email'],
  };
  const { container } = render(<HeaderRow {...props} />);
  expect(container).toMatchSnapshot();
});
