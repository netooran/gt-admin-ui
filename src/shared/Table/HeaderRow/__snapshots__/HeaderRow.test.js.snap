// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders header row with actions 1`] = `
<div>
  <tr>
    <th>
      name
    </th>
    <th>
      email
    </th>
    <th>
      actions
    </th>
  </tr>
</div>
`;

exports[`renders header row with select all 1`] = `
<div>
  <tr>
    <th>
      <input
        aria-label="Select all"
        type="checkbox"
      />
    </th>
    <th>
      name
    </th>
    <th>
      email
    </th>
  </tr>
</div>
`;

exports[`renders header row without actions 1`] = `
<div>
  <tr>
    <th>
      name
    </th>
    <th>
      email
    </th>
  </tr>
</div>
`;

exports[`renders header row without select all 1`] = `
<div>
  <tr>
    <th>
      name
    </th>
    <th>
      email
    </th>
  </tr>
</div>
`;
